### DISCUSSION TITLE
Lark.js - a port of Lark to Javascript

### DISCUSSION THREAD

**Discussion by erezsh** [lark-parser/lark#947] [Show and tell]
In the past few weeks I've been working on porting the standalone portion of Lark to Javascript. It's not entirely done yet, but I am already able to generate a JSON parser that works completely, including the use of a Transformer, and with all of lark's LALR features. It's about 75% faster than Lark on CPython.

The plan is to create a new repo, Lark.js, that is able to generate a standalone LALR parser (of about 10kb, before grammar). Future versions might include generating an Earley parser too.

The next step, now that the technical elements are mostly done, is to turn it into a package that follows the idioms and practices of the Javascript community. Which, I must admit, I'm not well immersed in.

So, I'm looking for someone who's a Javascript expert,  who can advise me on improving adoption, who can help me adapt the API, who can review the code I have so far,  and so on.

Please note that we have a budget for this! Due to a recent generous donation by Smore, we can afford to hire someone to do it, if his credentials justify it. So if you know someone who maintains a popular Open-source Javascript package and might be interested, please send him my way.

**Comment by swamidass** [comment]
I think you need to get it into NPM to have any adoption. It's also worth showing some benchmarks relative to other javascript parsers, and perhaps ensuring lark can work with their input formats too.
